src/wallet/wallet.cpp

```diff
@@ -260,6 +261,69 @@ std::shared_ptr<CWallet> LoadWalletInternal(WalletContext& context, const std::s
         return nullptr;
     }
 }
+
+class FastWalletRescanFilter
+{
+public:
+    FastWalletRescanFilter(const CWallet& wallet) : m_wallet(wallet)
+    {
+        // fast rescanning via block filters is only supported by descriptor wallets right now
+        assert(!m_wallet.IsLegacy());
+
+        // create initial filter with scripts from both active and non-active ScriptPubKeyMans
+        auto active_spkms{m_wallet.GetActiveScriptPubKeyMans()};
+        for (auto spkm : m_wallet.GetAllScriptPubKeyMans()) {
+            auto desc_spkm{dynamic_cast<DescriptorScriptPubKeyMan*>(spkm)};
+            AddScriptPubKeys(desc_spkm);
+            // save each active ScriptPubKeyMans's range end for possible future filter updates
+            if (active_spkms.find(spkm) != active_spkms.end()) {
+                m_last_range_ends.emplace(desc_spkm->GetID(), GetDescriptorRangeEnd(*desc_spkm));
+            }
+        }
```

Comment:
Can remove the `active_spkms` lookup by using `desc_spkm->IsRange()`.
E.g.
```c++
for (auto spkm : m_wallet.GetAllScriptPubKeyMans()) {
     auto desc_spkm{dynamic_cast<DescriptorScriptPubKeyMan*>(spkm)};
     AddScriptPubKeys(desc_spkm);
     // save each spkm range end for possible future filter updates
     if (desc_spkm->IsRange()) {
           m_last_range_ends.emplace(desc_spkm->GetID(), GetDescriptorRangeEnd(*desc_spkm));
     }
}
```

Reply:
Similar to the rationale in https://github.com/bitcoin/bitcoin/pull/25957#discussion_r969540182, I'd prefer to not extend the SPK-man interface with new methods unless absolutely needed, if I don't even know if it would be useful outside of this PR (which is "only" an optimization and doesn't add any functionality of its own). Only for saving one line of code in the ctor I don't think it's worth it.